### PR TITLE
Gjoranv/refactor ports  

### PR DESCRIPTION
**Pull request by gjoranv**
This is to allow disabling of the messageBus port without disabling admin rpc used by slobrok.

### CODE REVIEW

File: config-model/src/main/java/com/yahoo/vespa/model/container/Container.java [hunk 1]
```diff
@@ -239,8 +240,7 @@ public boolean requiresConsecutivePorts() {
      */
     public int getPortCount() {
         int httpPorts = (getHttp() != null) ? 0 : numHttpServerPorts + 2; // TODO remove +2, only here to keep irrelevant unit tests from failing.
-        int rpcPorts = (rpcServerEnabled()) ? numRpcServerPorts : 0;
-        return httpPorts + rpcPorts;
+        return httpPorts + numMessageBusPorts() + numRpcPorts();
```
Comment (bjorncs) [inline config-model/src/main/java/com/yahoo/vespa/model/container/Container.java]: Can the port count be extracted from `portsMeta`? This method and the getters below seems to duplicate the logic/info already stored in `portsMeta`.

**Review by bjorncs — APPROVED** [review]

File: config-model/src/main/java/com/yahoo/vespa/model/container/Container.java (hunk 1 above)
Comment (gjoranv) [reply config-model/src/main/java/com/yahoo/vespa/model/container/Container.java]: No, because getPortCount is called before the `portsMeta` is populated.

**Comment by gjoranv** [comment]
I'll look into the failing test in StandaloneContainer

**Comment by gjoranv** [comment]
Resolved by disabling messageBus when rpcServer is disabled in ContainerModelBuilder.

**Comment by gjoranv** [comment]
Merging to test this on factory.

**Pull request merged**